app/src/main-process/log.ts

```diff
@@ -78,12 +78,7 @@ function getLogger(): Promise<winston.LogMethod> {
 
     const logPath = getLogPath()
 
-    Fs.mkdir(logPath, (error) => {
-      if (error && error.code !== 'EEXIST') {
-        reject(error)
-        return
-      }
-
+    fileSystem.mkdirIfNeeded(logPath, (error) => { reject(error) }).then(() => {
```

Comment:
I'm not 100% across the wrapping of core Node APIs as promises, but I'll walk through how I understand it to work and see if we're on the same page.

I was under the impression that you could do only do this:

```ts
fileSystem.mkdir(path, (err) => {
  if (err) {
    // error handling
    return
  }
  // it worked
})
```

Or this with the core Node APIs (not the lack of the callback here):

```
try {
  await fileSystem.mkdir(path)
  // it worked
} catch (err) {
  // error handling
}
```

In both of these cases we don't specifically rethrow the error, so the promise is considered fulfilled - and the `then` callback would fire.

I think we should just stick to wrapping the callback API in a `Promise` here, like we've done in other places, as it lets us set the callback and also makes the `resolve` and `reject` cases explicit.